raiden/transfer/node.py

```diff
@@ -836,35 +840,35 @@ def handle_receive_transfer_refund_cancel_route(
 ) -> TransitionResult[ChainState]:
 
     new_secrethash = state_change.secrethash
-    current_payment_task = chain_state.payment_mapping.secrethashes_to_task[
+    current_payment_task = chain_state.payment_mapping[state_change.sender].secrethashes_to_task[
         state_change.transfer.lock.secrethash
     ]
-    chain_state.payment_mapping.secrethashes_to_task.update(
+    chain_state.payment_mapping[state_change.sender].secrethashes_to_task.update(
         {new_secrethash: copy.deepcopy(current_payment_task)}
     )
-    return subdispatch_to_paymenttask(chain_state, state_change, new_secrethash, storage)
+    return subdispatch_to_paymenttask(chain_state, state_change, state_change.transfer.target, new_secrethash, storage)
 
 
 def handle_receive_transfer_cancel_route(
     chain_state: ChainState, state_change: ReceiveTransferCancelRoute
 ) -> TransitionResult[ChainState]:
     return subdispatch_to_paymenttask(
-        chain_state, state_change, state_change.transfer.lock.secrethash
+        chain_state, state_change, state_change.transfer.target, state_change.transfer.lock.secrethash
     )
 
 
 def handle_receive_secret_request(
     chain_state: ChainState, state_change: ReceiveSecretRequest
 ) -> TransitionResult[ChainState]:
     secrethash = state_change.secrethash
-    return subdispatch_to_paymenttask(chain_state, state_change, secrethash)
+    return subdispatch_to_paymenttask(chain_state, state_change, chain_state.our_address, secrethash)
 
 
 def handle_receive_secret_request_light(
     chain_state: ChainState, state_change: ReceiveSecretRequestLight
 ) -> TransitionResult[ChainState]:
     secrethash = state_change.secrethash
-    return subdispatch_to_paymenttask(chain_state, state_change, secrethash)
+    return subdispatch_to_paymenttask(chain_state, state_change, state_change.recipient, secrethash)
```

Comment:
could you please explain why the upper function needs `our_address` but the lower one needs `recipient`?